app/src/main/java/org/jellyfin/androidtv/preference/PreferencesRepository.kt

```diff
@@ -25,11 +28,13 @@ class PreferencesRepository(
 		return store
 	}
 
-	suspend fun onSessionChanged() {
-		// Note: Do not run parallel as the server can't deal with that
-		// Relevant server issue: https://github.com/jellyfin/jellyfin/issues/5261
-		liveTvPreferences.update()
-		userSettingPreferences.update()
+	suspend fun onSessionChanged() = coroutineScope {
+		for (i in 1..100) {
+			awaitAll(
+				async { liveTvPreferences.update() },
+				async { userSettingPreferences.update() },
+			)
+		}
```

Comment:
this change runs the two updates in parallel instead of waterfall. I added the for loop to make sure  the bug is triggered as those requests can respond pretty fast so it doesn't always trigger the bug otherwise. The update function calls the `getDisplayPreferences()` endpoint on the server.